barretenberg/cpp/src/barretenberg/vm2/simulation/execution.hpp

```diff
@@ -74,11 +79,16 @@ class Execution : public ExecutionInterface {
                             const std::vector<Operand>& resolved_operands);
     std::vector<Operand> resolve_operands(const Instruction& instruction, const ExecInstructionSpec& spec);
 
+    void emit_context_snapshot(ContextInterface& context);
```

Comment:
@fcarreiro: Thoughts on a similar interface to this for emitting the context event, rather than it being a member function of the `Context` class?